python/cuml/filter/kalman_filter.pyx

```diff
@@ -14,492 +14,14 @@
-        algo = _get_algorithm_c_name(self.algorithm)
-
-        if isinstance(z, cudf.Series):
-            z_ptr = self._get_column_ptr(z)
-
-        elif isinstance(z, np.ndarray):
-            z_dev = rmm.to_device(z)
-            z_ptr = z.device_ctypes_pointer.value
-
-        elif cuda.devicearray.is_cuda_ndarray(z):
-            z_ptr = z.device_ctypes_pointer.value
-
-        else:
-            msg = "z format supported"
-            raise TypeError(msg)
-
-        if self.precision == 'single':
-
-            with nogil:
-
-                init_f32(var32,
-                         <int> dim_x,
-                         <int> dim_z,
-                         <Option> algo,
-                         <float*> _x_est_ptr,
-                         <float*> _x_up_ptr,
-                         <float*> _Phi_ptr,
-                         <float*> _P_est_ptr,
-                         <float*> _P_up_ptr,
-                         <float*> _Q_ptr,
-                         <float*> _R_ptr,
-                         <float*> _H_ptr,
-                         <void*> _ws_ptr,
-                         <size_t&> current_size)
-
-                update_f32(var32,
-                           <float*> z_ptr)
-
-        else:
-
-            with nogil:
-
-                init_f64(var64,
-                         <int> dim_x,
-                         <int> dim_z,
-                         <Option> algo,
-                         <double*> _x_est_ptr,
-                         <double*> _x_up_ptr,
-                         <double*> _Phi_ptr,
-                         <double*> _P_est_ptr,
-                         <double*> _P_up_ptr,
-                         <double*> _Q_ptr,
-                         <double*> _R_ptr,
-                         <double*> _H_ptr,
-                         <void*> _ws_ptr,
-                         <size_t&> current_size)
-
-                update_f64(var64,
-                           <double*> z_ptr)
-
-    def __setattr__(self, name, value):
-        if name in ["F", "x_up", "x", "P_up", "P", "Q", "H", "R", "z"]:
-            if (isinstance(value, cudf.DataFrame)):
-                val = numba_utils.row_matrix(value)
-
-            elif (isinstance(value, cudf.Series)):
-                val = value._column._data.mem
-
-            elif (isinstance(value, np.ndarray) or
-                  cuda.devicearray.is_cuda_ndarray(value)):
-                val = rmm.to_device(value)
-
-            super(KalmanFilter, self).__setattr__(name, val)
-
-        else:
-            super(KalmanFilter, self).__setattr__(name, value)
-
-
-def _get_algorithm_c_name(algorithm):
-    return {
-        'long': LongForm,
-        'short_explicit': ShortFormExplicit,
-        'short_implicit': ShortFormImplicit,
-    }[algorithm]
+        raise NotImplementedError(
+            """The Kalman filter is no longer provided due to issues with the
+            previous implementation. See cuML issue #1758 for more details.""")
```

Comment:
I would add perhaps that cuml < 0.12 is available with the old KF. One thing that hit me is whether we should do a deprecation warning in 0.13 and drop in 0.14 instead of plain drop "to follow protocol" so to speak?

Reply:
On the interesting side, in the issue #1758 it is described as deprecation then dropping as I mentioned